sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -628,6 +628,40 @@ public static ExecutorService addShutdownHookSafely(ExecutorService executorServ
         return executorService;
     }
 
+    /**
+     * Helper method that safely adds a {@link Runtime#addShutdownHook(Thread)} to the JVM that will run when the JVM is
+     * shutting down.
+     * <p>
+     * {@link Runtime#addShutdownHook(Thread)} checks for security privileges and will throw an exception if the proper
+     * security isn't available. So, if running with a security manager, setting
+     * {@code AZURE_ENABLE_SHUTDOWN_HOOK_WITH_PRIVILEGE} to true will have this method use access controller to add
+     * the shutdown hook with privileged permissions.
+     * <p>
+     * If {@code shutdownThread} is null, no shutdown hook will be added and this method will return null.
+     *
+     * @param shutdownThread The {@link Thread} that will shut down the
+     * @return The {@code executorService} that was passed in.
+     * @throws NullPointerException If {@code shutdownTimeout} is null.
+     * @throws IllegalArgumentException If {@code shutdownTimeout} is zero or negative.
+     */
+    @SuppressWarnings({"deprecation", "removal"})
+    public static Thread addRuntimeShutdownHookSafely(Thread shutdownThread) {
```

Comment:
We should just make this an overload of the other method and call this `addShutdownHookSafely`.

Reply:
nit: method name should be updated to match the other method that accepts ExecutorService.